File: backend/src/main/java/com/supplier/order/entity/OrderTrack.java
package com.supplier.order.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.supplier.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("order_track")
public class OrderTrack extends BaseEntity {

    private Long orderId;
    private Integer trackStatus;
    private LocalDateTime trackTime;
    private String trackRemark;
    private Long operator;
    private String operatorName;
}

File: frontend/src/api/orderTrack.ts
import { request } from '@/utils/request'

export interface OrderTrackVO {
  id: number
  orderId: number
  trackStatus: number
  trackTime: string
  trackRemark: string
  operator: number | null
  operatorName: string
}

export const orderTrackApi = {
  listByOrderId: (orderId: number | string) =>
    request.get<OrderTrackVO[], OrderTrackVO[]>(`/v1/order-tracks/order/${orderId}`),
}
